src/display/editor/stamp.js

```diff
@@ -0,0 +1,411 @@
+/* Copyright 2022 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AnnotationEditor } from "./editor.js";
+import { AnnotationEditorType } from "../../shared/util.js";
+import { StampAnnotationElement } from "../annotation_layer.js";
+
+/**
+ * Basic text editor in order to create a FreeTex annotation.
+ */
+class StampEditor extends AnnotationEditor {
+  #bitmap = null;
+
+  #bitmapId = null;
+
+  #bitmapPromise = null;
+
+  #bitmapUrl = null;
+
+  #canvas = null;
+
+  #observer = null;
+
+  #resizeTimeoutId = null;
+
+  static _type = "stamp";
```

Comment:
*You've probably already thought about this, but:* Once this lands, please don't forget to add the new "stamp"-type to the relevant telemetry code on the platform side.
 - https://searchfox.org/mozilla-central/rev/9a4666e63199bd1bcfc9095f6efec3488c358458/toolkit/components/pdfjs/content/PdfJsTelemetry.sys.mjs#25
 - https://searchfox.org/mozilla-central/rev/9a4666e63199bd1bcfc9095f6efec3488c358458/toolkit/components/pdfjs/metrics.yaml#17-21

Reply:
Yep it's on my todo.